crates/language/src/language_registry.rs

```diff
@@ -506,23 +507,25 @@ impl LanguageRegistry {
         self: &Arc<Self>,
         path: &Path,
         content: Option<&Rope>,
-        user_file_types: Option<&HashMap<Arc<str>, Vec<String>>>,
+        user_file_types: Option<&HashMap<Arc<str>, GlobSet>>,
     ) -> impl Future<Output = Result<Arc<Language>>> {
```

Comment:
Regarding the comment to remove the `Option` wrapper as seen [here](https://github.com/zed-industries/zed/pull/10960#discussion_r1582259019), I think it'll affect uses of the function where user_file_types can be empty as seen [here](https://github.com/zed-industries/zed/blob/a575ad8f37ee08accdc76098c37c4826c97f0365/crates/language/src/language_registry.rs#L502). In such cases, an empty hash map will need to be passed. So I have left the `Option` wrapper there for now.